Observer/TrackingSaveObserver.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+namespace Bolt\Boltpay\Observer;
+
+use Bolt\Boltpay\Helper\Api as ApiHelper;
+use Magento\Framework\Event\ObserverInterface;
+use Magento\Framework\Event\Observer;
+use Bolt\Boltpay\Helper\Config as ConfigHelper;
+use Magento\Framework\DataObjectFactory;
+use Bolt\Boltpay\Helper\Bugsnag;
+
+/**
+ * Class TrackingSaveObserver
+ *
+ * @package Bolt\Boltpay\Observer
+ */
+class TrackingSaveObserver implements ObserverInterface
+{
+    /**
+     * @var ConfigHelper
+     */
+    private $configHelper;
+
+    /**
+     * @var DataObjectFactory
+     */
+    private $dataObjectFactory;
+
+    /**
+     * @var ApiHelper
+     */
+    private $apiHelper;
+
+    /**
+     * @var Bugsnag
+     */
+    private $bugsnag;
+
+    /**
+     * @param ConfigHelper      $configHelper
+     * @param DataObjectFactory $dataObjectFactory
+     *
+     */
+    public function __construct(
+        ConfigHelper $configHelper,
+        DataObjectFactory $dataObjectFactory,
+        ApiHelper $apiHelper,
+        Bugsnag $bugsnag
+    ) {
+        $this->configHelper = $configHelper;
+        $this->dataObjectFactory = $dataObjectFactory;
+        $this->apiHelper = $apiHelper;
+        $this->bugsnag = $bugsnag;
+    }
+
+    /**
+     * @param Observer $observer
+     */
+    public function execute(Observer $observer)
+    {
+        try {
```

Comment:
lets use a feature switch to wrap all of this especially since we are listening to all trackevents.

we also probably want to track only bolt orders?

Reply:
agreed, is there an easy way to just track bolt orders?

Reply:
this is resolved apart from only tracking bolt orders. Not sure how that would work

Reply:
ended up checking for transaction_referenc to see if this was a bolt order